decorator.rb

```diff
@@ -0,0 +1,14 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```

Comment:
# UPDATE FOR THE PERSON CLASS

- Let's update the `Person` class to inherit from `Nameable` class. It should also have the method `correct_name`:

```ruby
class Person < Nameable

``` 
Then add the method
```ruby
def correct_name
       @name
    end
```
---

- Also kindly add the missing part to see your decoration in action:

![Screenshot from 2022-11-09 21-19-50](https://user-images.githubusercontent.com/59397159/200933817-d5ab716a-4403-4319-bd6c-0da6efada8c7.png)